tests/openstack/task.yaml

```diff
@@ -0,0 +1,101 @@
+summary: Test the openstack backend.
+
+systems: [ubuntu-22.04-64-devstack]
+
+kill-timeout: 30m
+
+prepare: |
+    if [ ! -f .spread-reuse.yaml ]; then
+        touch /run/spread-reuse.yaml
+        ln -s /run/spread-reuse.yaml .spread-reuse.yaml
+    fi
+
+    sudo -u stack -i /opt/stack/devstack/unstack.sh
+    sudo -u stack -i /opt/stack/devstack/stack.sh
+
+execute: |
+    # Check the instances which cannot be accessed are discarded
+    # The instance was created and the status has to be active to
+    # fail trying to access through ssh
+    spread openstack:cirros-64: -v -reuse -resend &> task.out || true
+    grep 'cannot find ready marker in console output for .*: timeout reached' task.out
+
+    # Check that the server is discarded during the spread execution
+    for i in $(seq 10); do
+        if [ -z "$(openstack server list)" ]; then
+            break
+        fi
+        sleep 1
+    done
+    test -z "$(openstack server list)"
+
+    # Check the error in case the network does not exist
+    spread openstack:cirros-64-wrong-network: -v -reuse -resend &> task.out || true
+    grep 'cannot find valid network with name "noexist"' task.out
+    test -z "$(openstack server list)"
```

Comment:
We have a system with very nice abstraction for composing test scenarios.  Is there a good reason for us to choose to cook all of them as a shell script inside a single task instead of using that composition to at least bundle closely related ideas together?

Reply:
Initially I though this but the problem I found is the time that devstack consumes to start (about 10 minutes), so if I create variants it will require more machines to get results. If makes sense I could move the scenarios to different variants and run them in parallel using more workers.